test/ably/restauth_test.py

```diff
@@ -2,67 +2,80 @@
 
 import logging
 import unittest
-import os 
+import responses
+import mock
+
+from os.path import join, abspath, dirname, normpath
 
 from ably import AblyRest
 from ably import Auth
 from ably import Options
+from ably.util.exceptions import AblyException
 
+log = logging.getLogger(__name__)
 
-from test.ably.restsetup import RestSetup
-
-test_vars = RestSetup.get_test_vars()
 
+class TestAuth(unittest.TestCase):
 
-log = logging.getLogger(__name__)
+    def _add_stats_responses(self):
+        with open(normpath(join(abspath(dirname(__file__)), '..', 'api_responses', 'stats.json'))) as f:
+            body = f.read()
+        responses.add(
+            responses.GET, 'https://rest.ably.io:443/stats',
+            body=body, status=200,
+            content_type='application/json',
+            adding_headers={
+                'access-control-expose-headers': 'Link',
+                'vary': 'Origin', 'connection': 'keep-alive',
+                'link': '<./stats?start=0&end=1431454230723&limit=100&unit=minute&'
+                        'direction=backwards&format=json&first_end=1431454230723>; rel="first", '
+                        '<./stats?start=0&end=1431454230723&limit=100&unit=minute&'
+                        'direction=backwards&format=json&first_end=1431454230723>; rel="current"',
+                'access-control-allow-credentials': 'true',
+                'date': 'Tue, 12 May 2015 18:10:30 GMT',
+                'access-control-allow-origin': '*'
+            })
+        responses.add(
+            responses.POST, 'https://rest.ably.io:443/keys/mykey.myvalue/requestToken',
+            body='{"error": {"statusCode": 401, "code": 40101, "message": "No keyName specified" }}',
```

Comment:
I found this: https://github.com/ably/ably-js/issues/41 yet could not fix the Python lib myself.

It looks like API has been changed at some point, and the outdated library returns 401 response.